stp_core/common/logging/TimeAndSizeRotatingFileHandler.py

```diff
@@ -30,11 +30,38 @@ def rotation_filename(self, default_name: str):
         maxIndex = -1
         for fileName in fileNames:
             if fileName.startswith(defaultFileName):
-                split = fileName.split(".")
-                try:
-                    index = int(split[-1] if len(split) > 0 else 0)
-                except ValueError:
-                    index = 0
+                index = self._file_index(fileName)
                 if index > maxIndex:
                     maxIndex = index
         return "{}.{}".format(default_name, maxIndex + 1)
+
+    @staticmethod
+    def _file_index(file_name):
+        split = file_name.split(".")
+        try:
+            return int(split[-1] if len(split) > 0 else 0)
+        except ValueError:
+            return 0
+
+    def getFilesToDelete(self):
+        """
+        Determine the files to delete when rolling over.
+
+        More specific than the earlier method, which just used glob.glob().
```

Comment:
1. This method is a slightly modified copy of the one in TimedRotatingFileHandler. Does the license allow this?

2. Add note that this is actually the same code that in TimedRotatingFileHandler, but with modifications (add comments right before them with description why we need them)
3. Update description, since there were no `glob.glob()` in our code.